File: client/unity/Assets/Scripts/Assembly-CSharp/BoardListOnScr.cs
public class BoardListOnScr : OnScreen
{
	private class IActionBL1 : IAction
	{
		public void perform()
		{
			Canvas.startWaitDlg();
			CasinoService.gI().joinAnyBoard();
		}
	}

	private class IActionBL3 : IAction
	{
		private BoardListOnScr p;

		public IActionBL3(BoardListOnScr p)
		{
			this.p = p;
		}

		public void perform()
		{
			p.doAskForBoardToGo();
		}
	}

	private class IActionBL4 : IAction
	{
		public void perform()
		{
			Canvas.startWaitDlg();
			GlobalService.gI().requestInfoOf(GameMidlet.avatar.IDDB);
		}
	}

	private class IActionToGo : IKbAction
	{
		private BoardListOnScr bscr;

		public IActionToGo(BoardListOnScr b)
		{
			bscr = b;
		}

		public void perform(string text)
		{
			me.boardIDToGo = int.Parse(text);
			me.doAskForPass();
		}
	}

	private class IActionPass : IKbAction
	{
		public void perform(string text)
		{
			CasinoService.gI().joinBoard(me.roomID, (sbyte)me.boardIDToGo, text);
			Canvas.endDlg();
		}
	}

	private class IActionXeng : IAction
	{
		public void perform()
		{
			TransMoneyDlg.gI().show();
		}
	}

	private class IActionJoinBoard : IKbAction
	{
		public void perform(string text)
		{
			BoardInfo boardInfo = (BoardInfo)me.boardList.elementAt(me.selected);
			CasinoService.gI().joinBoard(me.roomID, boardInfo.boardID, text);
			Canvas.load = 0;
		}
	}

	public static BoardListOnScr me;

	public static sbyte STYLE_2PLAYER = 0;

	public static sbyte STYLE_4PLAYER = 1;

	public static sbyte STYLE_5PLAYER = 2;

	public static sbyte type = STYLE_4PLAYER;

	public static FrameImage imgBoard;

	public static Image imgSoBan;

	public static Image imgLock;

	public Image imgTitleBoard;

	public Image imgNumPlayer;

	public Image imgPlaying;

	private int nBoardPerLine;

	private int defX;

	public MyVector boardList = new MyVector();

	public int xStart;

	public int yStart;

	public int countPaint;

	public sbyte roomID;

	private short wSmall;

	private Command cmdSellect;

	private Command cmdMenu;

	private Command cmdClose;

	public int cmtoY;

	public int cmy;

	public int cmdy;

	public int cmvy;

	public int cmyLim;

	public static Image imgSelectBoard;

	private int boardIDToGo;

	private int dis;

	private long timeDelay;

	private int pa;

	private int vY;

	private int dyTran;

	private int timePoint;

	private int count;

	private bool transY;

	private bool isGO;

	private int timeOpen;

	private int pyLast;

	private bool trans;

	private bool isG;

	public BoardListOnScr()
	{
		defX = 0;
		cmdSellect = new Command(T.selectt, 1);
		center = new Command(T.playNow, 5);
		right = new Command("Đến bàn", 4);
		addCmd(2, 1);
		wSmall = (short)(110 * AvMain.hd);
		nBoardPerLine = Canvas.w / wSmall + 1;
		if (nBoardPerLine * wSmall > Canvas.w - wSmall / 2)
		{
			nBoardPerLine--;
		}
		xStart = wSmall / 2;
		yStart = wSmall / 2 + 20 * AvMain.hd;
		yStart += 10;
		if (Canvas.w > nBoardPerLine * wSmall)
		{
			xStart = (Canvas.w - nBoardPerLine * wSmall) / 2 + wSmall / 2;
		}
	}

	public static BoardListOnScr gI()
	{
		return (me != null) ? me : (me = new BoardListOnScr());
	}

	public override void switchToMe()
	{
		repaint();
		selected = 0;
		loadImgBoard();
		isHide = true;
		RoomListOnScr.gI().roomList.removeAllElements();
		countPaint = 0;
		base.switchToMe();
	}

	public void loadImgBoard()
	{
		Canvas.paint.initImgBoard(type);
	}

	public new void close()
	{
		Canvas.startWaitDlg();
		doExitBoardList();
	}

	public override void commandTab(int index)
	{
		switch (index)
		{
		case 1:
			doJoinBoard();
			break;
		case 2:
			doExitBoardList();
			break;
		case 3:
			commandAction(1);
			break;
		case 4:
			doAskForBoardToGo();
			break;
		case 5:
			Canvas.startWaitDlg();
			CasinoService.gI().joinAnyBoard();
			break;
		}
	}

	public override void commandAction(int index)
	{
		switch (index)
		{
		case 1:
			Canvas.startWaitDlg(T.pleaseWait);
			CasinoService.gI().requestBoardList(roomID);
			break;
		case 3:
			Canvas.startWaitDlg();
			GlobalService.gI().requestInfoOf(GameMidlet.avatar.IDDB);
			break;
		case 4:
			doExitBoardList();
			break;
		}
	}

	private void doAskForBoardToGo()
	{
		ipKeyboard.openKeyBoard(T.goToBoard, ipKeyboard.NUMBERIC, string.Empty, new IActionToGo(this), false);
	}

	protected void doAskForPass()
	{
		ipKeyboard.openKeyBoard(T.ifPassword, ipKeyboard.PASS, string.Empty, new IActionPass(), false);
	}

	public void setXeng()
	{
		Canvas.startOKDlg("Hiện tại bạn không đủ Xèng để tham gia màn chơi, bạn có muốn nạp thêm Xèng không?", new IActionXeng());
	}

	protected void doJoinBoard()
	{
		BoardInfo boardInfo = (BoardInfo)boardList.elementAt(selected);
		if (MapScr.isNewVersion && boardInfo.money > GameMidlet.avatar.money[3])
		{
			gI().setXeng();
			return;
		}
		if (boardInfo.isPass)
		{
			ipKeyboard.openKeyBoard(T.ifPassword, ipKeyboard.NUMBERIC, string.Empty, new IActionJoinBoard(), false);
			return;
		}
		CasinoService.gI().joinBoard(roomID, boardInfo.boardID, string.Empty);
		Canvas.load = 0;
	}

	private void doExitBoardList()
	{
		Canvas.cameraList.close();
		CasinoService.gI().requestRoomList();
		Canvas.load = 0;
		RoomListOnScr.gI().switchToMe();
	}

	public override void paint(MyGraphics g)
	{
		Canvas.resetTrans(g);
		Canvas.paint.paintDefaultBg(g);
		onMainMenu.paintTitle(g, T.room + RoomListOnScr.title + " " + roomID, Canvas.w / 2, 30 * AvMain.hd - cmy);
		if (boardList.size() > 0 && Canvas.load == -1)
		{
			paintBoardList(g);
		}
		base.paint(g);
		Canvas.paintPlus2(g);
	}

	public void paintBoardList(MyGraphics g)
	{
		g.translate(xStart, yStart);
		g.translate(0f, -cmy);
		int num = cmy / wSmall * nBoardPerLine - nBoardPerLine;
		if (num < 0)
		{
			num = 0;
		}
		int num2 = num + Canvas.h / wSmall * nBoardPerLine + nBoardPerLine * 2 + nBoardPerLine;
		if (num2 > boardList.size())
		{
			num2 = boardList.size();
		}
		int width = imgTitleBoard.getWidth();
		int height = imgTitleBoard.getHeight();
		for (int i = num; i < num2 && i < boardList.size() && i < countPaint; i++)
		{
			int num3 = i % nBoardPerLine * wSmall;
			int num4 = i / nBoardPerLine * wSmall;
			BoardInfo boardInfo = (BoardInfo)boardList.elementAt(i);
			if (i == selected && !isHide)
			{
				g.drawImage(imgSelectBoard, num3, num4 + 10 * AvMain.hd, 3);
			}
			imgBoard.drawFrame(boardInfo.nPlayer, num3, num4 + 10 * AvMain.hd, 0, 3, g);
			g.drawImage(imgTitleBoard, num3, num4 - imgBoard.frameHeight / 2 - 5 * AvMain.hd, 3);
			Canvas.menuFont.drawString(g, string.Empty + boardInfo.boardID, num3 - width / 2 + 10 * AvMain.hd, num4 - imgBoard.frameHeight / 2 - 5 * AvMain.hd - Canvas.menuFont.getHeight() / 2, 2);
			Canvas.numberFont.drawString(g, boardInfo.strMoney, num3 + 8 * AvMain.hd, num4 - imgBoard.frameHeight / 2 - 5 * AvMain.hd - Canvas.numberFont.getHeight() / 2, 2);
			if ((type == STYLE_4PLAYER && boardInfo.maxPlayer < 4) || (type == STYLE_5PLAYER && boardInfo.maxPlayer < 5))
			{
				g.drawImage(imgNumPlayer, num3 + width / 2, num4 - imgBoard.frameHeight / 2 - 5 * AvMain.hd, 3);
				Canvas.menuFont.drawString(g, string.Empty + boardInfo.maxPlayer, num3 + width / 2, num4 - imgBoard.frameHeight / 2 - 5 * AvMain.hd - Canvas.menuFont.getHeight() / 2 + 1, 2);
			}
			if (boardInfo.isPlaying)
			{
				g.drawImage(imgPlaying, num3, num4 + 10 * AvMain.hd, 3);
			}
			if (boardInfo.isPass)
			{
				g.drawImage(imgLock, num3 + wSmall / 2 - 10 * AvMain.hd, num4 + wSmall / 2 - 14 * AvMain.hd, 3);
			}
		}
	}

	public override void updateKey()
	{
		base.updateKey();
		count++;
		if (Canvas.isPointerClick)
		{
			pyLast = Canvas.pyLast;
			isG = false;
			if (Canvas.isPoint(0, 0, Canvas.w, dis))
			{
				if (vY != 0)
				{
					isG = true;
				}
				pa = cmtoY;
				timeDelay = count;
				trans = true;
			}
			Canvas.isPointerClick = false;
		}
		if (!trans)
		{
			return;
		}
		int num = pyLast - Canvas.py;
		int i = pyLast - Canvas.py;
		pyLast = Canvas.py;
		long num2 = count - timeDelay;
		if (Canvas.isPointerDown)
		{
			if (count % 2 == 0)
			{
				dyTran = Canvas.py;
				timePoint = count;
			}
			vY = 0;
			if (Math.abs(num) < 5 && Math.abs(i) < 5)
			{
				int num3 = (Canvas.py - (yStart - wSmall / 2) + cmtoY) / wSmall;
				int num4 = (Canvas.px - (xStart - wSmall / 2)) / wSmall;
				if (num3 >= 0 && num3 < boardList.size() && Canvas.isPoint(xStart - wSmall / 2, 0, Canvas.w - (xStart - wSmall / 2) * 2, dis))
				{
					selected = num3 * nBoardPerLine + num4;
				}
			}
			if (CRes.abs(Canvas.dy()) >= 5 && CRes.abs(Canvas.dx()) >= 5)
			{
				isHide = true;
			}
			else if (num2 > 3 && num2 < 8)
			{
				int num5 = (Canvas.py - (yStart - wSmall / 2) + cmtoY) / wSmall;
				if (num5 >= 0 && num5 < boardList.size() && !isG)
				{
					isHide = false;
				}
			}
			if (cmtoY < 0 || cmtoY > cmyLim)
			{
				cmtoY = pa + num / 2;
				pa = cmtoY;
			}
			else
			{
				cmtoY = pa + num / 2;
				pa = cmtoY;
			}
			cmy = cmtoY;
		}
		if (!Canvas.isPointerRelease || !Canvas.isPoint(0, 0, Canvas.w, dis))
		{
			return;
		}
		isG = false;
		int num6 = count - timePoint;
		int num7 = dyTran - Canvas.py;
		if (CRes.abs(num7) > 40 && num6 < 10 && cmtoY > 0 && cmtoY < cmyLim)
		{
			vY = num7 / num6 * 10;
		}
		timePoint = -1;
		if (Math.abs(num) < 5 && Math.abs(i) < 5)
		{
			if (num2 <= 4)
			{
				isHide = false;
				timeOpen = 5;
			}
			else if (!isHide)
			{
				click();
			}
		}
		trans = false;
		Canvas.isPointerRelease = false;
	}

	private void click()
	{
		cmdSellect.perform();
	}

	public void setBoardList(MyVector boardList)
	{
		this.boardList = boardList;
		dis = Canvas.hCan - PaintPopup.hButtonSmall;
		int num = boardList.size() / nBoardPerLine;
		if (boardList.size() % nBoardPerLine > 0)
		{
			num++;
		}
		cmyLim = num * wSmall - dis;
		if (cmyLim < 0)
		{
			cmyLim = 0;
		}
	}

	public void setCam()
	{
		int num = boardList.size() / nBoardPerLine;
		if (boardList.size() % nBoardPerLine != 0)
		{
			num++;
		}
		yStart = 110 * AvMain.hd;
	}

	public override void update()
	{
		if (timeOpen > 0)
		{
			timeOpen--;
			if (timeOpen == 0)
			{
				click();
			}
		}
		if (countPaint < boardList.size())
		{
			countPaint++;
		}
		moveCamera();
	}

	public void moveCamera()
	{
		if (vY != 0)
		{
			if (cmy < 0 || cmy > cmyLim)
			{
				vY -= vY / 4;
				cmy += vY / 20;
				if (vY / 10 <= 1)
				{
					vY = 0;
				}
			}
			if (cmy < 0)
			{
				if (cmy < -dis / 2)
				{
					cmy = -dis / 2;
					cmtoY = 0;
					vY = 0;
				}
			}
			else if (cmy > cmyLim)
			{
				if (cmy < cmyLim + dis / 2)
				{
					cmy = cmyLim + dis / 2;
					cmtoY = cmyLim;
					vY = 0;
				}
			}
			else
			{
				cmy += vY / 10;
			}
			cmtoY = cmy;
			vY -= vY / 10;
			if (vY / 10 == 0)
			{
				vY = 0;
			}
		}
		else if (cmy < 0)
		{
			cmtoY = 0;
		}
		else if (cmy > cmyLim)
		{
			cmtoY = cmyLim;
		}
		if (cmy != cmtoY)
		{
			cmvy = cmtoY - cmy << 2;
			cmdy += cmvy;
			cmy += cmdy >> 4;
			cmdy &= 15;
		}
	}
}

File: client/java/src/avt/BoardListOnScr.java
package avt;

import java.io.IOException;
import java.util.Vector;
import javax.microedition.lcdui.Graphics;
import javax.microedition.lcdui.Image;
import main.Canvas;
import main.GameMidlet;

public final class BoardListOnScr extends MyScreen {
   public static BoardListOnScr me;
   public static byte STYLE_2PLAYER = 0;
   public static byte STYLE_4PLAYER = 1;
   public static byte STYLE_5PLAYER = 2;
   public static byte type;
   public static FrameImage imgBoard;
   private static Image imgTitleBoard;
   private static Image imgNumPlayer;
   private static Image imgPlay;
   private static Image imgLock;
   private int nBoardPerLine;
   Vector boardList;
   private int p;
   private int q;
   public byte roomID;
   private short wSmall;
   private Command cmdSellect;
   public static Image imgSelectBoard;
   int j;

   public static BoardListOnScr gI() {
      return me == null ? (me = new BoardListOnScr()) : me;
   }

   public final void switchToMe() {
      OnScreen.disableVirtualKey();
      MyScreen.repaint();
      super.selected_ = 0;
      Canvas.paint.setValue(type);
      if (imgTitleBoard == null) {
         try {
            imgTitleBoard = Image.createImage(T.getPath() + "/on/imgkhungsoban.on");
            imgNumPlayer = Image.createImage(T.getPath() + "/on/imgNumPlayer.on");
            imgPlay = Image.createImage(T.getPath() + "/on/imgPlay.on");
            imgLock = Image.createImage(T.getPath() + "/on/imgLock.on");
         } catch (IOException var2) {
            var2.printStackTrace();
         }
      }

      Canvas.load = 1;
      super.isHide_ = true;
      GameMidlet.avatar.ableShow = false;
      super.switchToMe();
   }

   public BoardListOnScr() {
      this.cmdSellect = new Command(T.selectt, 1);
      super.right = new Command(T.close, 2);
      if (Canvas.stypeInt != 0) {
         super.center = new Command(T.strongest, 5);
      } else {
         super.center = this.cmdSellect;
      }

      super.left = new Command(T.menu, 6);
      this.wSmall = (short)(110 * AvMain.hd);
      if (Canvas.stypeInt == 1) {
         this.wSmall = 95;
      } else if (Canvas.stypeInt == 0) {
         this.wSmall = (short)(Canvas.w / 4);
         if (this.wSmall < 70) {
            this.wSmall = (short)(Canvas.w / 3);
         }

         if (Canvas.w < 180) {
            this.wSmall = (short)(Canvas.w / 2);
         }
      }

      this.nBoardPerLine = Canvas.w / this.wSmall + 1;
      if (this.nBoardPerLine * this.wSmall > Canvas.w - this.wSmall / 2) {
         --this.nBoardPerLine;
      }

      this.p = this.wSmall / 2;
      this.q = this.wSmall / 2;
      this.q += 10;
      if (Canvas.w > this.nBoardPerLine * this.wSmall) {
         this.p = (Canvas.w - this.nBoardPerLine * this.wSmall) / 2 + this.wSmall / 2;
      }

   }

   public final void close() {
      Canvas.startWaitDlg();
      doExitBoardList();
   }

   public final void commandTab(int var1, int var2) {
      switch (var1) {
         case 1:
            BoardInfo var4 = (BoardInfo)this.boardList.elementAt(super.selected_);
            if (MapScr.isNewVersion && var4.money > GameMidlet.avatar.money[3]) {
               gI().setXeng();
            } else {
               if (!var4.isPass) {
                  CasinoService.gI().joinBoard(this.roomID, var4.boardID, "");
                  Canvas.startWaitDlg();
                  return;
               }

               Canvas.inputDlg.setImg(T.ifPassword, new IActionJoinBoard(this), 2);
            }
            break;
         case 2:
            doExitBoardList();
            return;
         case 3:
            this.commandActionPointer(1, -1);
            return;
         case 4:
            this.doAskForBoardToGo();
            return;
         case 5:
            Canvas.startWaitDlg();
            CasinoService.gI().joinAnyBoard();
            return;
         case 6:
            Vector var3;
            (var3 = new Vector()).addElement(new Command(T.strongest, 5));
            var3.addElement(new Command("Đến bàn", 6));
            var3.addElement(MapScr.gI().f);
            var3.addElement(new Command(T.viewMyInfo, 7));
            Menu.gI().startAt(var3, 0);
      }

   }

   public final void commandActionPointer(int var1, int var2) {
      switch (var1) {
         case 1:
            Canvas.startWaitDlg(T.pleaseWait);
            CasinoService.gI().requestBoardList(this.roomID);
            return;
         case 3:
            Canvas.startWaitDlg();
            GlobalService.gI().requestInfoOf(GameMidlet.avatar.IDDB);
            return;
         case 4:
            doExitBoardList();
            return;
         case 5:
            Canvas.startWaitDlg();
            CasinoService.gI().joinAnyBoard();
            return;
         case 6:
            this.doAskForBoardToGo();
            return;
         case 7:
            Canvas.startWaitDlg();
            GlobalService.gI().requestInfoOf(GameMidlet.avatar.IDDB);
         case 2:
         default:
      }
   }

   private void doAskForBoardToGo() {
      Canvas.inputDlg.setImg(T.goToBoard, new IActionToGo(this), 3);
   }

   protected final void doAskForPass() {
      Canvas.inputDlg.setImg(T.banChiCo, new IActionPass(this), 0);
   }

   public final void setXeng() {
      Canvas.startOKDlg("Hiện tại bạn không đủ Xèng để tham gia màn chơi, bạn có muốn nạp thêm Xèng không?", new IActionXeng(this));
   }

   private static void doExitBoardList() {
      Canvas.cameraList.isShow = false;
      CasinoService.gI().requestRoomList();
      Canvas.startWaitDlg();
   }

   public final void paint(Graphics var1) {
      Canvas.resetTrans(var1);
      RoomListOnScr.paintRoomList(var1, "Phòng " + RoomListOnScr.title + " " + this.roomID);
      this.paintBoardList(var1);
      OnScreen.paintTitle(var1, super.left, super.center, super.right);
      Canvas.paintPlus2(var1);
   }

   private void paintBoardList(Graphics var1) {
      var1.translate(this.p, this.q);
      var1.translate(0, -CameraList.cmtoY);
      int var2;
      if ((var2 = CameraList.cmtoY / this.wSmall * this.nBoardPerLine - this.nBoardPerLine) < 0) {
         var2 = 0;
      }

      int var3;
      if ((var3 = var2 + Canvas.h / this.wSmall * this.nBoardPerLine + (this.nBoardPerLine << 1) + this.nBoardPerLine) > this.boardList.size()) {
         var3 = this.boardList.size();
      }

      for(var2 = var2; var2 < var3; ++var2) {
         int var4 = var2 % this.nBoardPerLine * this.wSmall;
         int var5 = var2 / this.nBoardPerLine * this.wSmall;
         BoardInfo var6 = (BoardInfo)this.boardList.elementAt(var2);
         if ((!Canvas.isKeyBoard || !super.isHide_) && var2 == super.selected_) {
            var1.drawImage(imgSelectBoard, var4, var5, 3);
         }

         imgBoard.drawFrame(var6.nPlayer, var4, var5, 0, 3, var1);
         var1.drawImage(imgTitleBoard, var4 - this.wSmall / 4, var5 - 30 * AvMain.hd, 3);
         Canvas.smallFontYellow.drawString(var1, "" + var6.boardID, var4 - this.wSmall / 4, var5 - 30 * AvMain.hd - AvMain.hSmall / 2, 2);
         if (var6.money > 0) {
            Canvas.smallFontYellow.drawString(var1, var6.strMoney, var4, var5 - 30 * AvMain.hd - AvMain.hSmall / 2, 2);
         }

         if (type == STYLE_4PLAYER && var6.maxPlayer < 4) {
            var1.drawImage(imgNumPlayer, var4 + this.wSmall / 4, var5 - 30 * AvMain.hd, 3);
            Canvas.smallFontRed.drawString(var1, "" + var6.maxPlayer, var4 + this.wSmall / 4, var5 - 30 * AvMain.hd - AvMain.hSmall / 2, 2);
         }

         if (var6.isPlaying) {
            var1.drawImage(imgNumPlayer, var4 - this.wSmall / 4, var5 + this.wSmall / 3, 3);
            var1.drawImage(imgPlay, var4 - this.wSmall / 4, var5 + this.wSmall / 3, 3);
         }

         if (var6.isPass) {
            var1.drawImage(imgNumPlayer, var4 + this.wSmall / 4, var5 + this.wSmall / 3, 3);
            var1.drawImage(imgLock, var4 + this.wSmall / 4, var5 + this.wSmall / 3, 3);
         }
      }

   }

   public final void updateKey() {
      if (OnScreen.isOngame && Canvas.stypeInt != 0) {
         Canvas.paint.updateKeyOn(super.left, super.center, super.right);
      } else {
         super.updateKey();
      }

   }

   public final void setBoardList(Vector var1) {
      this.boardList = var1;
   }

   public final void init() {
      int var1 = this.boardList.size() / this.nBoardPerLine;
      if (this.boardList.size() % this.nBoardPerLine != 0) {
         ++var1;
      }

      this.q = 100 * AvMain.hd;
      if (Canvas.w < 200) {
         this.q = 50;
      }

      Canvas.cameraList.setInfo(this.p - this.wSmall / 2, this.q - this.wSmall / 2, this.wSmall, this.wSmall, this.nBoardPerLine * this.wSmall, var1 * this.wSmall + 10, this.nBoardPerLine * this.wSmall, Canvas.h - (this.q - this.wSmall / 2) - 4, this.boardList.size());
   }

   public final void setSelected(int var1, boolean var2) {
      if (var2 && super.selected_ == var1 && this.cmdSellect != null) {
         this.cmdSellect.perform();
      }

      if (var1 >= 0 && var1 < this.boardList.size()) {
         super.setSelected(var1, var2);
      }

   }

   public final void update() {
   }

   static {
      type = STYLE_4PLAYER;
   }
}
